packages/squiggle-lang/src/rescript/ReducerProject/ReducerProject.res

```diff
@@ -112,6 +112,16 @@ module Private = {
     handleNewTopology(project)
   }
 
+  let removeSource = (project: t, sourceId: string): bool => {
+    if Belt.Array.length(getDependents(project, sourceId)) == 0 {
+      removeContinues(project, sourceId)
+      Belt.Map.String.remove(project["items"], sourceId)->T.Private.setFieldItems(project, _)
+      true
+    } else {
+      false
```

Comment:
true/false is surprising, should it be `result` or throw an exception? We're switching to the exceptions in many other places for the performance, and I think exceptions are fine, even for external APIs (or maybe especially for external APIs, since people in JS land are not often familiar with `result` objects pattern, and JS doesn't force you to check the returned values).

Reply:
Hmmm. OTOH, "you tried to delete the item but someone depends on it" is a pretty common scenario in Observable. I wonder what the consequences of this are...

Can you tell Observable that the cell can't be deleted? Probably not from react hook cleanup (or maybe not at all). This looks like a problem; stale invisible objects left behind in a Project. So maybe we should delete the item anyway?

Reply:
Yeah! Good point. I've fixed it so that it always works even if it puts the project in an invalid state.